src/ui/buttons/script.jsx

```diff
@@ -166,11 +163,21 @@ function getComponentScript() : () => void {
 }
 
 type ScriptProps = {|
-    nonce : ?string
+    nonce : ?string,
+    // eslint-disable-next-line flowtype/no-weak-types
+    buttonAnimation : any
 |};
 
-export function Script({ nonce } : ScriptProps) : ElementNode {
+export function Script({ nonce, buttonAnimation } : ScriptProps) : ElementNode {
+    const htmlForScript = `
+        const scriptFns = ${ getComponentScript().toString() }
+        const animation = ${ buttonAnimation && buttonAnimation.fn ? buttonAnimation.fn.toString() : null }
```

Comment:
Doesn't need to be a ternary. Only use a ternary when you have relevant return values for both cases
(buttonAnimation && buttonAnimation.fn) || buttonAnimation.fn.toString()

Reply:
I think we need the Ternary because we need to convert the function into string and the || buttonAnimation.fn.toString() only gets evaluated when (buttonAnimation && buttonAnimation.fn) is false